content/guides/clj_datatype_constructs.adoc

```diff
@@ -0,0 +1,153 @@
+= Understanding Clojure's Polymorphism
+Ikuru Kanuma
+2017-07-20
+:type: guides
+:toc: macro
+:icons: font
+
+ifdef::env-github,env-browser[:outfilesuffix: .adoc]
+
+== Goals of this guide
+
+Clojue supports several constructs for speaking to the Java world
```

Comment:
Important because the proxy macro was created first specifically for speaking to the Java world via extending classes and implementing interfaces.

Later came protocols/reify/deftype/defrecord which are not created for Java interop, but instead are Clojure's own constructs for polymorphic dispatch.

As a newbie the overlap of proxy and reify were confusing until I learned the above distinction.